hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -718,8 +747,10 @@ public void stop() {
         return;
       }
       balancerRunning = false;
-      currentBalancingThread.interrupt();
-      currentBalancingThread.join(1000);
+      if (Thread.currentThread().getId() != currentBalancingThread.getId()) {
+        currentBalancingThread.interrupt();
+        currentBalancingThread.join(0);
+      }
 
       // allow garbage collector to collect balancing thread
       currentBalancingThread = null;
```

Comment:
maybe we could ignore catching InterruptedException here , `finally` is enough.
no matter `stop` is called by balancing thread itself or command line,  the lock will be hold, thus when `join`, no `interrupt()` could be called.

Reply:
Could `interrupt` be called on balancer from a test method?

Reply:
i think in test method , `containerbalancer#stop` may be called , but `interrupt` will never be called directly because `currentBalancingThread` is private.

Reply:
Okay, agreed!